lib/Service/UserMigrationService.php

```diff
@@ -276,6 +347,69 @@ public function getMigrators(): array {
 		return $migrators;
 	}
 
+	/**
+	 * @return null|UserExport|UserImport
+	 *
+	 * @throws UserMigrationException
+	 */
+	public function getCurrentJob(IUser $user) {
+		// TODO merge export and import entities?
+
+		try {
+			$exportJob = $this->exportMapper->getBySourceUser($user->getUID());
+		} catch (DoesNotExistException $e) {
+			// Allow this exception as this just means the user has no export jobs queued currently
+		}
+
+		try {
+			$importJob = $this->importMapper->getByTargetUser($user->getUID());
+		} catch (DoesNotExistException $e) {
+			// Allow this exception as this just means the user has no import jobs queued currently
+		}
+
+		if (!empty($exportJob) && !empty($importJob)) {
+			throw new UserMigrationException('A user export and import job cannot be queued or run at the same time');
+		}
+
+		$job = $exportJob ?? $importJob;
+
+		if (empty($job)) {
+			return null;
+		}
+
+		return $job;
+	}
+
+	/**
+	 * @param UserExport|UserImport $job
+	 *
+	 * @throws UserMigrationException
+	 */
+	public function cancelJob($job): void {
```

Comment:
We should try to account for bogus status where the job is not in the list but it exists in the mapper. Will canceling work in this case? We need to give users a way out of the deadlock.